File: Source/JavaScript/model/interactions/SetStateAction.ts
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

import { BindingExpression } from '../common';
import { InteractionAction } from './InteractionAction';

/**
 * Writes screen state.
 */
export interface SetStateAction extends InteractionAction {
    target: string;
    value: BindingExpression;
}

export const SetStateActionPropertyNames: (keyof SetStateAction)[] = ['target', 'value', 'kind'];

File: Source/DotNET/Model/Interactions/SetStateAction.cs
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

using Cratis.Scene.Model.Common;

namespace Cratis.Scene.Model.Interactions;

/// <summary>
/// Writes screen state.
/// </summary>
/// <param name="Target">The declared state name being written.</param>
/// <param name="Value">The value to write.</param>
public record SetStateAction(string Target, BindingExpression Value) : InteractionAction
{
    /// <inheritdoc/>
    public override InteractionActionKind Kind => InteractionActionKind.SetState;
}
